scalr/data_source_current_run.go

```diff
@@ -121,7 +122,7 @@ func dataSourceScalrCurrentRunRead(d *schema.ResourceData, meta interface{}) err
 	log.Printf("[DEBUG] Read workspace of run: %s", runID)
 	workspace, err := scalrClient.Workspaces.ReadByID(ctx, run.Workspace.ID)
 	if err != nil {
-		if err == scalr.ErrResourceNotFound {
+		if errors.Is(err, scalr.ErrResourceNotFound{}) {
```

Comment:
```
The Unwrap method
Go 1.13 introduces new features to the errors and fmt standard library packages to simplify working with errors that contain other errors. The most significant of these is a convention rather than a change: an error which contains another may implement an Unwrap method returning the underlying error. If e1.Unwrap() returns e2, then we say that e1 wraps e2, and that you can unwrap e1 to get e2.

Following this convention, we can give the QueryError type above an Unwrap method that returns its contained error:

func (e *QueryError) Unwrap() error { return e.Err }
The result of unwrapping an error may itself have an Unwrap method; we call the sequence of errors produced by repeated unwrapping the error chain.
```
Якщо нова структура буде повертати старий ерор. Можна обійтись без створення нової структури у перевірках. 
```scalr.ErrResourceNotFound = errors.New("resource not found")```
```errors.Is(err, scalr.ErrResourceNotFound)```

Reply:
cc @lbn

Reply:
так, можна обiйтись взагалi без структури, це буде валiдний еррор i перевiрки типу .Is/.As будуть працювати. Структуру залишив як базовий 404 ерор, щоб вiд нього вже створювати специфiчнi пiд ресурси i запаковувати помилку.